src/item.cpp

```diff
@@ -3168,6 +3175,15 @@ int item::chip_resistance( bool worst ) const
     return res;
 }
 
+bool item::mod_damage( int qty, damage_type dmg )
+{
+    damage_ = std::max( std::min( damage_ + qty, max_damage() ), min_damage() );
+    if( qty > 0 ) {
+        on_damage( dmg, qty );
+    }
+    return damage_ < max_damage();
```

Comment:
Inverted logic? Description states it returns whether the item should be destroyed.